pkg/work/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -208,3 +215,45 @@ func (m *ManifestWorkController) applyAppliedManifestWork(ctx context.Context, w
 	_, err = m.appliedManifestWorkPatcher.PatchSpec(ctx, appliedManifestWork, requiredAppliedWork.Spec, appliedManifestWork.Spec)
 	return appliedManifestWork, err
 }
+
+func (m *ManifestWorkController) manifestWorkCompletion(
+	ctx context.Context, controllerContext factory.SyncContext, manifestWork *workapiv1.ManifestWork,
+) (bool, error) {
+	complete := meta.FindStatusCondition(manifestWork.Status.Conditions, workapiv1.ManifestComplete)
```

Comment:
why not IsStatusConditionTrue?

Reply:
Because it uses last transition time from the condition to determine if TTL has elapsed for deletion